core/templates/pages/topic-editor-page/subtopic-editor/subtopic-editor-tab.component.html

```diff
@@ -1,48 +1,52 @@
 <div class="subtopic-content">
-  <p class="oppia-mobile-back-to-parent" ng-click="$ctrl.navigateToTopicEditor()">
+  <p class="oppia-mobile-back-to-parent"
+     (click)="navigateToTopicEditor()">
     <i class="fa fa-angle-left"></i>
     <span>Back to Topic</span>
   </p>
   <div class="subtopic-header-text">
-    <span class="topic-name" ng-click="$ctrl.navigateToTopicEditor()"><[$ctrl.topic.getName()]> / </span>
-    <span class="subtopic-name"><[$ctrl.editableTitle]></span>
+    <span class="topic-name" (click)="navigateToTopicEditor()">{{ topic?.getName() }} / </span>
```

Comment:
In what scenario is topic undefined?

Reply:
topic is not undefined, `?` (safe navigation operator) is used to ensure that UI doesn't break anywhere. See this for more info https://docs.angular.lat/guide/template-expression-operators#the-safe-navigation-operator----and-null-property-paths. Sometimes the UI renders before the function calls, which results in unwanted console errors, to prevent this we're using safe navigation operator.

Reply:
What does the UI look like when topic is undefined? Should we use an ngIf in a top level element in the UI so that we only show the user the content when topic is defined (in case it looks weird / errors are shown to the user)?

Reply:
errors are only shown in the console for the time `topic` is undefined, but the UI still renders properly. Just to prevent the below console errors we can either use a `ngIf` or `?.` operator, both of them work fine. 
![image](https://user-images.githubusercontent.com/60519303/195259624-8c00e97f-63c5-42e1-8c74-1c4195c69f97.png)
We're also using this `?.` operator in other templates in the codebase. Stating from the docs itself
```
Sometimes however, null values in the property path may be OK under certain circumstances, especially when the value starts out null but the data arrives eventually.
```
Ig `*ngIf` in top level element will stop loading the whole component and wait for `topic` property but `?.` renders the UI first and changes it eventually.

Reply:
Hm, is the UI ready for interaction when the topic is undefined? e.g. Can there be a scenario where the user interacts with the component before the editor is initialised? This can cause unexpected server errors. To avoid such a case, I think it would be safer to use an ngIf.

Reply:
done, added ngIf check to be extra safe